hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/S3ABlockOutputStream.java

```diff
@@ -608,6 +611,16 @@ private long putObject() throws IOException {
             builder.putOptions,
         false);
 
+    PutObjectRequest.Builder maybeModifiedPutIfAbsentRequest = putObjectRequest.toBuilder();
+    Map<String, String> optionHeaders = builder.putOptions.getHeaders();
+
+    if (optionHeaders != null && optionHeaders.containsKey(IF_NONE_MATCH_HEADER)) {
+        maybeModifiedPutIfAbsentRequest.overrideConfiguration(
```

Comment:
as mentioned below as well, think we should upgrade SDK and then use the new .ifNoneMatch(). 

Also I would recommend you move all of this logic into a new private method in this class.

Reply:
actually, I would move this logic to RequestFactoryImpl. But you will need to add in a flag to the method signature, "isConditonalPutEnabled", and only add the if-none-match header in when sConditonalPutEnabled is true. 

Basically, you only want to add  ` if-none-match` when the PUT is  coming from S3ABlockOutputStream.close() and not from anywhere else.